worker/remoterelations/remoterelations.go

```diff
@@ -420,125 +449,198 @@ func (w *remoteApplicationWorker) relationChanged(
+	// The export was done when the relation was registered.
+	token, err = w.localModelFacade.GetToken(w.localModelUUID, names.NewApplicationTag(applicationName))
+	if err != nil {
+		return errors.Annotatef(err, "getting token for application %v from offering model", applicationName)
+	}
+	w.relationInfo.localApplicationId = params.RemoteEntityId{ModelUUID: w.localModelUUID, Token: token}
+	return nil
+}
+
+// processNewConsumingRelation starts the sub-workers necessary to listen and publish
+// local unit settings changes, and watch and consume remote unit settings changes.
+func (w *remoteApplicationWorker) processNewConsumingRelation(
+	key string,
+	relations map[string]*relation,
+	remoteRelation *params.RemoteRelation,
+) error {
+	// We have not seen the relation before, make
+	// sure it is registered on the offering side.
+	w.relationInfo.localEndpoint = remoteRelation.Endpoint
+	w.relationInfo.remoteEndpointName = remoteRelation.RemoteEndpointName
+
+	applicationTag := names.NewApplicationTag(remoteRelation.ApplicationName)
+	relationTag := names.NewRelationTag(key)
+	applicationId, remoteApplictionId, relationId, err := w.registerRemoteRelation(applicationTag, relationTag)
+	if err != nil {
+		return errors.Annotatef(err, "registering application %v and relation %v", remoteRelation.ApplicationName, relationTag.Id())
 	}
-	relationUnitsWatcher, err := newRelationUnitsWatcher(
+	w.relationInfo.localApplicationId = applicationId
+
+	// Start a watcher to track changes to the units in the relation in the local model.
+	localRelationUnitsWatcher, err := w.localModelFacade.WatchLocalRelationUnits(key)
+	if err != nil {
+		return errors.Annotatef(err, "watching local side of relation %v", relationTag.Id())
+	}
+
+	// localUnitSettingsFunc converts relations units watcher results from the local model
+	// into settings params using an api call to the local model.
+	localUnitSettingsFunc := func(changedUnitNames []string) ([]params.SettingsResult, error) {
+		relationUnits := make([]params.RelationUnit, len(changedUnitNames))
+		for i, changedName := range changedUnitNames {
+			relationUnits[i] = params.RelationUnit{
+				Relation: relationTag.String(),
+				Unit:     names.NewUnitTag(changedName).String(),
+			}
+		}
+		return w.localModelFacade.RelationUnitSettings(relationUnits)
+	}
+	localUnitsWorker, err := newRelationUnitsWorker(
 		relationTag,
-		w.relationInfo.applicationId,
+		applicationId,
 		w.macaroon,
-		remoteRelationId,
+		relationId,
 		localRelationUnitsWatcher,
-		w.relationChanges,
-		w.facade,
+		w.localRelationChanges,
+		localUnitSettingsFunc,
+	)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	if err := w.catacomb.Add(localUnitsWorker); err != nil {
+		return errors.Trace(err)
+	}
+
+	// Start a watcher to track changes to the units in the relation in the remote model.
+	remoteRelationUnitsWatcher, err := w.remoteModelFacade.WatchRemoteRelationUnits(relationId)
```

Comment:
Ah, ok - so `WatchRemoteRelationUnits` is called on the remote facade, but it returns information about changes to local-for-it units. Maybe it should be called `WatchRelationUnits` instead?